openmdao/drivers/tests/test_pyoptsparse_driver.py

```diff
@@ -2270,6 +2270,171 @@ def test_singular_jac_tol(self):
 
         prob.run_driver()
 
+    def test_multiple_constraints(self):
+        p = om.Problem()
+
+        exec = om.ExecComp(['y = x**2',
+                            'z = a + x**2'],
+                        a={'shape': (1,)},
+                        y={'shape': (101,)},
+                        x={'shape': (101,)},
+                        z={'shape': (101,)})
+
+        p.model.add_subsystem('exec', exec)
+
+        p.model.add_design_var('exec.a', lower=-1000, upper=1000)
+        p.model.add_objective('exec.y', index=50)
+        p.model.add_constraint('exec.z', indices=[0], equals=25)
+        p.model.add_constraint('exec.z', indices=[-1], lower=20, alias="ALIAS_TEST")
+
+        p.driver = om.pyOptSparseDriver()
+        p.driver.options['optimizer'] = OPTIMIZER
+
+        p.setup()
+
+        p.set_val('exec.x', np.linspace(-10, 10, 101))
+
+        p.run_driver()
+
+        assert_near_equal(p.get_val('exec.z')[0], 25)
+        assert_near_equal(p.get_val('exec.z')[50], -75)
+
+    def test_multiple_constraints_no_alias(self):
+        p = om.Problem()
+
+        exec = om.ExecComp(['y = x**2',
+                            'z = a + x**2'],
+                        a={'shape': (1,)},
+                        y={'shape': (101,)},
+                        x={'shape': (101,)},
+                        z={'shape': (101,)})
+
+        p.model.add_subsystem('exec', exec)
+
+        p.model.add_design_var('exec.a', lower=-1000, upper=1000)
+        p.model.add_objective('exec.y', index=50)
+        p.model.add_constraint('exec.z', indices=[0], equals=25)
+
+        msg = "Constraint 'exec.z' already exists. Use the 'alias' argument to apply a second constraint"
+        with self.assertRaises(RuntimeError) as msg:
+            p.model.add_constraint('exec.z', indices=[-1], lower=20)
+
+    def test_single_constraint_with_alias(self):
+        p = om.Problem()
+
+        exec = om.ExecComp(['y = x**2',
+                            'z = a + x**2'],
+                        a={'shape': (1,)},
+                        y={'shape': (101,)},
+                        x={'shape': (101,)},
+                        z={'shape': (101,)})
+
+        p.model.add_subsystem('exec', exec)
+
+        p.model.add_design_var('exec.a', lower=-1000, upper=1000)
+        p.model.add_objective('exec.y', index=50)
+        p.model.add_constraint('exec.z', indices=[-1], lower=20, alias="ALIAS_TEST")
+
+        p.driver = om.pyOptSparseDriver()
+        p.driver.options['optimizer'] = OPTIMIZER
+
+        msg = "Alias 'ALIAS_TEST' is not needed when only adding one constraint to model"
+        with self.assertRaises(RuntimeError) as msg:
+            p.setup()
+
+    def test_obj_and_con_same_var_different_indices(self):
+
+        import openmdao.api as om
```

Comment:
We no longer need to import anything in these tests, since moving all docs to notebooks.